includes/api/class-wc-admin-rest-onboarding-profile-controller.php

```diff
@@ -0,0 +1,330 @@
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_items( $request ) {
+		$query_args      = $this->prepare_objects_query( $request );
+		$onboarding_data = get_option( 'wc_onboarding_profile', array() );
+		$update          = update_option( 'wc_onboarding_profile', array_merge( $onboarding_data, $query_args ) );
+
+		if ( $update ) {
+			$result = array(
+				'status'  => 'success',
+				'message' => __( 'Onboarding profile data has been updated.', 'woocommerce-admin' ),
+			);
+		} else {
+			$result = array(
+				'status'  => 'error',
+				'message' => __( 'There was an error updating the onboarding profile data.', 'woocommerce-admin' ),
+			);
+		}
+
+		$response = $this->prepare_item_for_response( $result, $request );
+		$data     = $this->prepare_response_for_collection( $response );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare objects query.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		$args       = array();
+		$properties = self::get_profile_properties();
+
+		foreach ( $properties as $key => $property ) {
+			if ( isset( $request[ $key ] ) ) {
+				$args[ $key ] = $request[ $key ];
+			}
+		}
+
+		/**
+		 * Filter the query arguments for a request.
+		 *
+		 * Enables adding extra arguments or setting defaults for a post
+		 * collection request.
+		 *
+		 * @param array           $args    Key value array of query var to query value.
+		 * @param WP_REST_Request $request The request used.
+		 */
+		$args = apply_filters( 'woocommerce_rest_onboarding_profile_object_query', $args, $request );
+
+		return $args;
+	}
+
+
+	/**
+	 * Prepare the data object for response.
+	 *
+	 * @param object          $item Data object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$data     = $this->add_additional_fields_to_object( $item, $request );
+		$data     = $this->filter_response_by_context( $data, 'view' );
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filter the list returned from the API.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param array            $item     The original item.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_onboarding_profile', $response, $item, $request );
+	}
+
+	/**
+	 * Get onboarding profile properties.
+	 *
+	 * @return array
+	 */
+	public static function get_profile_properties() {
+		$properties = array(
+			'account_type'    => array(
+				'type'        => 'string',
+				'description' => __( 'Account type used for JetPack.', 'woocommerce-admin' ),
+				'context'     => array( 'view' ),
+				'readonly'    => true,
+			),
+			'store_address_1' => array(
```

Comment:
Part of the discussion in Slack involved the store address. I think we can probably just save this using the normal WC settings API -- and just use this profile endpoint for all the other fields.

Reply:
I had mixed feelings on this since what we could collect here might be different from something later set in WC settings, but I think in retrospect it's probably not necessary to maintain two copies of the store address information.

I dropped these properties, but do you think we should merge in these store setting fields in the GET request here or make a separate API call to settings if we need to grab them?

Reply:
Good question. Looking at the designs again, I don't actually think we need the address information in the other steps, so we can leave them out of the GET request for now. I was thinking maybe `country` since we will be showing prices of extensions, but we would just be using the user's WP language for that.

If we do end of needing data in the profiler though, I have no issues adding it here later since this is an internal endpoint.